pkg/destroy/aws/aws.go

```diff
@@ -101,17 +102,24 @@ func (o *ClusterUninstaller) validate() error {
 
 // Run is the entrypoint to start the uninstall process
 func (o *ClusterUninstaller) Run() error {
+	_, err := o.RunWithContext(context.Background())
+	return err
+}
+
+// RunWithContext runs the uninstall process with a context.
+// The first return is the list of ARNs for resources that could not be destroyed.
+func (o *ClusterUninstaller) RunWithContext(ctx context.Context) (notDeleted []string, returnErr error) {
```

Comment:
All consumers vendor us, so I'd rather have this new signature for `Run` and drop the old signature.  Consumers can update, and add Todo or background context, when they vendor-bump.

Reply:
That would mean that the interface would change for *all* platforms, where the destroyers for the other platforms would not honor the context. Can we have that as the end goal, but wait until I roll out changes to more platforms?

Reply:
Ah, makes sense :+1: